ocaml/xapi/xapi_pci_helpers.ml

```diff
@@ -172,3 +173,62 @@ let get_host_pcis () =
 let igd_is_whitelisted ~__context pci =
   let vendor_id = Db.PCI.get_vendor_id ~__context ~self:pci in
   List.mem vendor_id !Xapi_globs.igd_passthru_vendor_whitelist
+
+let is_pci_hidden_cmdline ~__context ~self =
+  let cmdline =
+    match Unixext.read_lines ~path:"/proc/cmdline" with
+    | [x] ->
+        x
+    | _ ->
+        failwith "Unable to read cmdline"
+  in
+  let device = Db.PCI.get_pci_id ~__context ~self in
+  let elems = String.split_on_char ' ' cmdline in
```

Comment:
This isn't how the parameters should be split, as per `man proc`:

> For processes which are still running, the command-line arguments appear in this file in the same layout as they do in process memory: If the process is well-behaved, it is a set of strings separated by null bytes ('\0'), with a further null byte after the last string.

This translates to `String.split_on_char '\000' cmdline`

Reply:
I agree. You can easily verify this: `cat /proc/self/cmdline`. This further suggests that this PR was not testes.

Reply:
Hmm I tested and it worked, and now I've tested with `\000` and it doesn't. :thinking:

Reply:
and if i cat the file there are spaces:
```
[11:07 localhost ~]# cat /proc/cmdline 
root=LABEL=root-bdijol ro nolvm hpet=disable console=hvc0 console=tty0 quiet vga=785 splash plymouth.ignore-serial-consoles xen-pciback.hide=(0000:01:00.1)(0000:0d:00.0)
```

It looks like the null byte is used in `/proc/<pid>/cmdline` though :thinking:

Reply:
You are right. One is for the kernel and the other for other processes. And they use different conventions, which is a bit maddening.

Reply:
it is yeah ^^'

Reply:
I don't know how it presented in the proc fs here, but usually splitting parameters by spaces is suspect because they can contain spaces